specs/emip-007/emip-007.md

```diff
@@ -0,0 +1,218 @@
+  "ConceptualWallet": Array<{
+    "ConceptualWalletId": number,
+    "CoinType": number,
+    "Name": string,
+    "NetworkId": number,
+    /** should not be stored in the wallet. Only used to show checksum to user when decrypting */
+    "Checksum": null | {
+      "ImagePart": string,
+      "TextPart": string,
+    }
+  }>,
+}
+```
+
+### Encryption
+
+The `walletInfo` section should be encrypted using the same algorithm as described by [EMIP3](https://github.com/Emurgo/EmIPs/blob/master/specs/emip-003.md) using a *decryption password*.
+
+*Note:* Although the ordering of the JSON keys doesn't matter for encryption, one SHOULD remove any un-neeeded whitespace to shrink the payload size.
+
+## Rationale
+
+### Why is part of the format encrypted?
+
+All identifiable information should be encrypted for maximum safety. However, information like wallet name and checksum do not reveal sensitive information and help the user know what is contained in the file (potentially also reminding them what the *decryption password* is).
+
+**Note**: we cannot encrypt with the user's spending password, as no spending password is available in cases like hardware wallets or read-only wallets.
+
+### What about user-labeled data?
+
+We don't export memos / labels or similar information since there is no way to comprehensively include them all in this specification (recall: different wallets could handle different kinds of user-labeled data). These should be specified as a different CIP and probably synced using a cloud storage provider instead of a wallet export.
+
+### What about transaction history?
+
+We don't need to include transaction history as it can be inferred from the blockchain
+
+### What is `AddressMapping` and why do we need it?
+
+A single derivation path can actually define multiple different address types. For example, in Cardano Shelley, a single derivation can define both "base" addresses and "enterprise" addresses. We therefore need a one-to-many mapping from the derivation to the addresses. The addresses need to be in a standalone table since rows in the address required for other metadata may not belong to your wallet.
+
+### How to handle multiple currencies from the same recovery phrase
+
+This format doesn't allow generating multiple currencies from a single recovery phrase. That would require another level above *ConceptualWallet*. This is because *ConceptualWallet* has to store information like the network ID for all its public derivers (which is a currency-specific concept).s
+
+### Why are addresses included if they can be derived from the public deriver?
+
+I include the addresses as although they can be derived when a public key is present, they cannot be derived in the balance tracker case. Additionally, it's possible in the future that some new metadata that needs to be tracked contains a foreign key to an address. Therefore, it's easier to just add them in the wallet export.
```

Comment:
I'm open to opinions on this. My concern is that it make cause a lot of compatibility issues down the line to include these